client/src/services/api/hooks/codeReview/codeReviewUtils.ts

```diff
@@ -45,5 +53,109 @@ export class CodeReviewMetricsUtils {
       pullRequests,
       errorCount: data.errorCount,
     };
-  };
+  }
+
+  static getFilteredPullRequestsByReviewers(
+    data: Omit<IFetchedRawCodeReviewResponse, "count" | "filteredCount"> | undefined,
+    filters: IContextClientFilterSquad[],
+  ): IFetchedCodeReviewResponse | undefined {
+    if (!data) {
+      return;
+    }
+
+    const { reviewersMap } = this.#getSquadsUserIdsMap(filters);
+
+    const pullRequests = data.pullRequests.map((pullRequest) => {
+      const filteredVotesTimeline = this.#filterVotesTimelineByReviewers(pullRequest.votesTimeline, reviewersMap);
+      const votes = this.#getPullRequestVotes(filteredVotesTimeline);
+
+      const filteredVotesHistoryTimeline = this.#filterVotesTimelineByReviewers(
+        pullRequest.votesHistoryTimeline,
+        reviewersMap,
+      );
+      const votesHistory = this.#getPullRequestVotes(filteredVotesHistoryTimeline);
+
+      const filteredThreads = this.#getFilteredThreads(pullRequest.threads, reviewersMap, pullRequest.authorId);
+
+      return {
+        id: pullRequest.id,
+        title: pullRequest.title,
+        status: pullRequest.status,
+        createdBy: pullRequest.createdBy,
+        authorId: pullRequest.authorId,
+        creationDate: pullRequest.creationDate,
+        closedDate: pullRequest.closedDate,
```

Comment:
Can we spread the `pullRequest` and then add the property which are needed ? Reason whenever we add new data in backend this file needs change